apps/graphslam-engine/graphslam-engine_app.cpp

```diff
@@ -262,292 +163,120 @@ int main(int argc, char **argv)
-			if (system::strCmpI(edge_reg, "CICPCriteriaERD")) { // CICPGooodnessNRD - CICPCriteriaERD
-				// Initialize the CGraphSlamEngine class
-				CGraphSlamEngine
-					<
-					CNetworkOfPoses2DInf,
-					CICPCriteriaNRD<CNetworkOfPoses2DInf>,
-					CICPCriteriaERD<CNetworkOfPoses2DInf>,
-					CLevMarqGSO<CNetworkOfPoses2DInf>
-					>
-					graph_engine(
-							ini_fname,
-							rawlog_fname,
-							ground_truth_fname,
-							!disable_visuals.getValue());
-				graph_engine.execGraphSlam();
-			}
-			else if (system::strCmpI(edge_reg, "CLoopCloserERD")) { // CFixedIntervalsNRD - CICPCriteriaERD
-				// Initialize the CGraphSlamEngine class
-				CGraphSlamEngine
-					<
-					CNetworkOfPoses2DInf,
-					CFixedIntervalsNRD<CNetworkOfPoses2DInf>,
-					CLoopCloserERD<CNetworkOfPoses2DInf>,
-					CLevMarqGSO<CNetworkOfPoses2DInf>
-					>
-					graph_engine(
-							ini_fname,
-							rawlog_fname,
-							ground_truth_fname,
-							!disable_visuals.getValue());
-				graph_engine.execGraphSlam();
-			}
-			else if (system::strCmpI(edge_reg, "CEmptyERD")) { // CICPGooodnessNRD - CEmptyERD
-
-				// Initialize the CGraphSlamEngine class
-				CGraphSlamEngine
-					<
-					CNetworkOfPoses2DInf,
-					CICPCriteriaNRD<CNetworkOfPoses2DInf>,
-					CEmptyERD<CNetworkOfPoses2DInf>,
-					CLevMarqGSO<CNetworkOfPoses2DInf>
-					>
-					graph_engine(
-							ini_fname,
-							rawlog_fname,
-							ground_truth_fname,
-							!disable_visuals.getValue());
-				graph_engine.execGraphSlam();
-			}
+		logger.logFmt(LVL_WARN, "Finished graphslam execution.");
+
+		//
+		// Postprocessing
+		//
+
+		logger.logFmt(LVL_INFO, "Generating overall report...");
+		graphslam_engine.generateReportFiles(graphslam_handler.output_dir_fname);
+		// save the graph and the 3DScene 
+		if (graphslam_handler.save_graph) {
+			std::string save_graph_fname = 
+				graphslam_handler.output_dir_fname +
+				"/" +
+				graphslam_handler.save_graph_fname;
+			graphslam_engine.saveGraph(&save_graph_fname);
+		}
+		if (!disable_visuals.getValue() && graphslam_handler.save_3DScene) {
+			std::string save_3DScene_fname = 
+				graphslam_handler.output_dir_fname +
+				"/" +
+				graphslam_handler.save_3DScene_fname;
+
+			graphslam_engine.save3DScene(&save_3DScene_fname);
+		}
 
+		// get the occupancy gridmap that was built
+		if (graphslam_handler.save_gridmap) {
+			COccupancyGridMap2D gridmap;
+			graphslam_engine.getOccupancyGridMap2D(&gridmap);
+			gridmap.saveMetricMapRepresentationToFile(
+					graphslam_handler.output_dir_fname +
+					"/" +
+					graphslam_handler.save_gridmap_fname);
 		}
+
+
 		////////////////////////////////////////////////////////////////////////
 
 	}
 	catch (exception& e) {
-		setConsoleColor(CONCOL_RED, true);
-		cerr << "Program finished with an exception!" << endl;
-		setConsoleColor(CONCOL_NORMAL, true);
+		setConsoleColor(CONCOL_RED,true);
+		cerr << "Program finished for an exception!!" << endl;
+		setConsoleColor(CONCOL_NORMAL,true);
 
```

Comment:
This console color stuff could be now switched to COutputLogger's output, for simplicity, right?

MRPT_LOG_ERROR("xxx");
or alike.